app/controllers/api/base_controller.rb

```diff
@@ -0,0 +1,16 @@
+module Api
+  class BaseController < ApplicationController
+    respond_to :json
+
+    SUCCESS = 'SUCCESS'.freeze
+    ERROR = 'ERROR'.freeze
+
+    def confirm_two_factor_authenticated_for_api
+      return if user_fully_authenticated?
+      render json: { error: 'user is not fully authenticated',
+                     status: ERROR,
+                   },
+             status: :unauthorized,
```

Comment:
For the syntax error, maybe just use the explicit parens?

```suggestion
      render(
        json: { error: 'user is not fully authenticated', status: ERROR },
        status: :unauthorized,
      )
```

Reply:
I feel like `render` is usually use without parens (since its an effect rather than an expression), so I think we could leave it paren-less and drop that trailing comma
```suggestion
      render json: { error: 'user is not fully authenticated',
                     status: ERROR,
                   },
             status: :unauthorized
```